pkg/cvo/sync_worker.go

```diff
@@ -668,8 +668,10 @@ func (w *SyncWorker) apply(ctx context.Context, payloadUpdate *payload.Update, w
 		return nil
 	})
 	if len(errs) > 0 {
-		err := cr.Errors(errs)
-		return err
+		if err := cr.Errors(errs); err != nil {
+			return err
+		}
+		return errs[0]
```

Comment:
So with this change we will always return an error. either summarized or err[0], even if all errors were context errors. Whereas before we could possibly be ignoring DeadlineExceeded errors, which is incorrect, but this also will not ignore Canceled either. Am I interpreting this correctly and is the latter intended?

Reply:
yup.  Neither `DeadlineExceeded` nor `Canceled` should result in us claiming the update completed.